### PR TITLE
Library Project  (JavaScript Section)

### PR DESCRIPTION
**Pull request by ClintonEnyinna**
We implemented the following features: 

- A user can add a new book to the library
- He/She can delete one or many books
- Can update the read status

### CODE REVIEW

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team  :clap: 

**Kindly check out my comments below;**

According to the Odin instructions, we need to add `remove button` on each book
image

According to the Odin instructions, we need to add `toggle button` on each book
image

image

It would be nice to add little styling, as you know this project would be added to your portfolio 👍 

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by bafiam** [comment]
Hey, Thanks for the feedback.
- Styling has been implemented

> **For the button, we improved the project by adding additional functionality. We used a checkbox which is more like a button in our case making the library more dynamic in that a user is able to delete a single selected entry as well able to delete more than one selected entry. In addition, we reused the same checkbox, to allow the user to toggle back and forth between a read status of "yes and no "; try it out. Also as an additional feature, user is able to toggle back and forth with more that one entry u know in the case of big data sets**. 

Thank you.

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    if (getMyInput[index].checked) {
+      const rowId = getMyInput[index].id;
+      delete myLibrary[rowId - 1];
+    }
+  }
+  const temp = myLibrary.filter((i) => i);
+
+  localStorage.setItem('myLibrary', JSON.stringify(temp));
+  window.location.reload();
+});
```
Comment (shloch) [inline js/main.js]: There's a good reason on using JAVASCRIPT, one of them is to not reload page unnecessarily. Ideally you should call the `render()` function after deleting data (Figure out a way to make that work)

### no reloads

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        myLibrary[rowId - 1].read = 'No';
+      } else {
+        myLibrary[rowId - 1].read = 'Yes';
+      }
+    }
+  }
+
+  localStorage.setItem('myLibrary', JSON.stringify(myLibrary));
+  window.location.reload();
+});
```
Comment (shloch) [inline js/main.js]: Ideally you should call the `render()` function after EDITING data (Figure out a way to make that work)

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  const getMyInput = document.querySelectorAll('.myinput');
+  for (let index = 0; index < getMyInput.length; index += 1) {
+    if (getMyInput[index].checked) {
+      const rowId = getMyInput[index].id;
+      if (myLibrary[rowId - 1].read === 'Yes') {
+        myLibrary[rowId - 1].read = 'No';
+      } else {
+        myLibrary[rowId - 1].read = 'Yes';
+      }
+    }
```
Comment (shloch) [inline js/main.js]: From the requirements, you had to put the `DELETE` button BESIDE each book.
Also, still from requirements, there's this javascript `data-attribute` (from a table display), that  gives you the `INDEX` of a book. This your algorithm does a search on all books to determine which one is checked. If there're 1000 books, for every delete, you run a search, this is very inefficient. On top of that you have a page reload.

([read about Data-attributes here](https://css-tricks.com/a-complete-guide-to-data-attributes/))
Inside your `render()` function when looping through books to generate them on a table, you add a delete button beside each book and pass `rowIndex` to that delete function (you get the idea)
 [[see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/remove_requirement.png)]

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+getDeleteBtn.addEventListener('click', () => {
+  const getMyInput = document.querySelectorAll('.myinput');
+  for (let index = 0; index < getMyInput.length; index += 1) {
+    if (getMyInput[index].checked) {
+      const rowId = getMyInput[index].id;
+      delete myLibrary[rowId - 1];
+    }
+  }
+  const temp = myLibrary.filter((i) => i);
```
Comment (shloch) [inline js/main.js]: Check my explanation above, regarding this search approach..it's inefficient
[[ see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/remove_requirement.png) ]

**Review by shloch — CHANGES_REQUESTED** [review]
## hello @ClintonEnyinna , @bafiam 

Nice job guys, It all works out fine........and properly

### [optional changes]

- Apply some controls to make sure that `pages` entered is always a number
Screen Shot 2020-04-24 at 12 13 50 AM
(This is not specified in requirements, but will be good to have)

### [required changes]
See observations below :

**Review by jofer86 — APPROVED** [review]
## :wave: :wave: Hello!, I'm Jorge, and  I'm here to review this Pull Request, let's take a look shall we :eye: :eye: :eye: 
- Changes marked as **[OPTIONAL]** are not hard requirements but are nice to have.
- Great job at addressing the required changes, with the implementations that you did the app runs more efficiently as you are leveraging the correct properties to do the required actions. Great job!!!
# `status: approved` :+1: :rocket: :+1: 
--------------------------------------------------------------------
Cheers, 
Jorge Fernando Rincón Figueroa.
Technical Support Engineer.
**Slack: @jofer86**

**Pull request merged**